phones/models.py

```diff
@@ -171,6 +171,24 @@ class RelayNumber(models.Model):
         max_length=6, default=vcard_lookup_key_default, unique=True
     )
     enabled = models.BooleanField(default=True)
+    remaining_minutes = models.IntegerField(
+        blank=True, null=True, default=settings.MAX_MINUTES_PER_BILLING_CYCLE
+    )
+    remaining_texts = models.IntegerField(
+        blank=True, null=True, default=settings.MAX_TEXTS_PER_BILLING_CYCLE
+    )
+    calls_forwarded = models.IntegerField(blank=True, null=True, default=0)
+    calls_blocked = models.IntegerField(blank=True, null=True, default=0)
+    texts_forwarded = models.IntegerField(blank=True, null=True, default=0)
+    texts_blocked = models.IntegerField(blank=True, null=True, default=0)
+
+    @property
+    def calls_and_texts_forwarded(self):
+        return self.calls_forwarded + self.texts_forwarded
```

Comment:
I appreciate you trying `null=True`, but it requires making the code that uses the nullable columns more careful. In `calls_and_texts_forwarded`, `self.calls_forwarded` can be `null`, and `self.texts_forwarded` can be null. `mypy` can catch this, if you add an annotation so it will check it rather than skip the function:

```python
    @property
    def calls_and_texts_forwarded(self) -> int:
        return self.calls_forwarded + self.texts_forwarded
```

which will result in the errors:

```
phones/models.py:187: error: Unsupported operand types for + ("int" and "None")  [operator]
phones/models.py:187: error: Unsupported operand types for + ("None" and "int")  [operator]
phones/models.py:187: error: Unsupported left operand type for + ("None")  [operator]
phones/models.py:187: note: Both left and right operands are unions
```

The fix:

```python
    @property
    def calls_and_texts_forwarded(self) -> int:
        return (self.calls_forwarded or 0) + (self.texts_forwarded or 0)
```

You'll need to add similar annotations and fixes to all the places that use ``calls_forwarded``, ``calls_blocked``, ``texts_forwarded``, ``texts_blocked``. I'm not sure if annotations will help with the API however, and some `null` values might get sent to the frontend.

Alternatively, set `null=False` and move the ugliness to the migrations.

Reply:
Okay I updated the models to drop the blank/null values and squashed it into new `phones/migrations/0021_add_relaynumber_stats_20220913_1959.py` migration with default values in the migration too.

Be sure to migrate BACKWARD before you try this.